### PR TITLE
Misc core team cleanups

### PR DESCRIPTION
**Pull request by NotMyFault**
Aligns with https://github.com/orgs/jenkinsci/teams/core/repositories

The deleted files match the archived repositories.

### CODE REVIEW

File: permissions/component-jenkins-test-harness-htmlunit.yml [hunk 1]
```diff
@@ -5,6 +5,4 @@ cd:
   enabled: true
 paths:
   - "org/jenkins-ci/main/jenkins-test-harness-htmlunit"
-developers:
-  - "drulli"
-  - "@core"
+developers: []
```
Comment (NotMyFault) [inline permissions/component-jenkins-test-harness-htmlunit.yml:8]: Fully switched to CD.

File: permissions/plugin-design-library.yml [hunk 2]
```diff
@@ -6,7 +6,6 @@ issues:
 paths:
   - "io/jenkins/plugins/design-library"
 developers:
-  - "@core"
```
Comment (NotMyFault) [inline permissions/plugin-design-library.yml:9]: We're using CD here, nor is this a core component or tool.

File: permissions/component-jenkins-test-harness-htmlunit.yml (hunk 1 above)
Comment (halkeye) [reply permissions/component-jenkins-test-harness-htmlunit.yml:8]: Btw A plugin with 0 developers listed is considered up for adoption

Comment (NotMyFault) [reply permissions/component-jenkins-test-harness-htmlunit.yml:8]: This is a library, not a plugin. You can't find it on plugins.jenkins.io, but we can add Ulli back, if needed.

Comment (NotMyFault) [inline permissions/component-jenkins-test-harness-htmlunit.yml:8]:
```suggestion
developers:
  - "drulli"
```

File: permissions/pom-archetypes.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+name: "archetypes"
+github: "jenkinsci/archetypes"
```
Comment (timja) [inline permissions/pom-archetypes.yml:3]: this is deployed to maven central why are you adding it here?

File: permissions/component-jenkins-test-harness-htmlunit.yml [hunk 4]
```diff
@@ -7,4 +7,3 @@ paths:
   - "org/jenkins-ci/main/jenkins-test-harness-htmlunit"
 developers:
   - "drulli"
-  - "@core"
```
Comment (timja) [inline permissions/component-jenkins-test-harness-htmlunit.yml:10]: why remove from here? even if using CD should still be able to release from here as a backup and this is also used for security contacts

File: permissions/pom-archetypes.yml (hunk 3 above)
Comment (NotMyFault) [reply permissions/pom-archetypes.yml]: Oic, yeah, 🤦🏻 me

Gonna remove it

**Review by timja — APPROVED** [review]

**Pull request merged**